### PR TITLE
Lower limit from API 21 to API 19

### PR DESCRIPTION
**Pull request by romainguy**
This was requested by an internal application. API 19 is when OpenGL
ES 3.0 support was added so there is no good reason for us to not
support this API level. The only trick is to avoid referring to the
glTexStorage2DMultisample symbol directly as it only exists in 3.1.

### CODE REVIEW

**Comment by romainguy** [comment]
@pixelflinger can you look at `OpenGLDriver` and `PlatformEGL`? I didn't want to use `eglGetProcAddress` from `OpenGLDriver`. Let me know if you'd like to do this differently. See the comments in the code for why I have to do this (basically we can lower our API level if we avoid referencing this one symbol directly).

**Comment by romainguy** [comment]
You're right, this was fixed in EGL 1.5 only. Doing a dlsym() is going to be painful. Apparently I also broke iOS.

**Comment by romainguy** [comment]
@pixelflinger PTAL

**Comment by romainguy** [comment]
@pixelflinger On API 19 I get a crash in our `ashmem` implementation:

```
09-26 20:07:14.472 1173-1173/? I/DEBUG: pid: 2538, tid: 2538, name: lament.textured  >>> com.google.android.filament.textured <<<
09-26 20:07:14.472 1173-1173/? I/DEBUG: signal 8 (SIGFPE), code -6 (SI_TKILL), fault addr 000009ea
09-26 20:07:14.612 1173-1173/? I/DEBUG:     eax 0fc45e85  ebx b7727fe4  ecx 0000005f  edx 00000000
09-26 20:07:14.612 1173-1173/? I/DEBUG:     esi 00000000  edi b7704004
09-26 20:07:14.612 1173-1173/? I/DEBUG:     xcs 00000073  xds 0000007b  xes 0000007b  xfs 00000000  xss 0000007b
09-26 20:07:14.612 1173-1173/? I/DEBUG:     eip b770eac8  ebp b642be70  esp bfb38680  flags 00210246
09-26 20:07:14.612 1173-1173/? I/DEBUG: backtrace:
09-26 20:07:14.612 1173-1173/? I/DEBUG:     #00  pc 00006ac8  /system/bin/linker (__dl__Z19dlsym_linear_lookupPKcPP6soinfoS2_+216)
09-26 20:07:14.612 1173-1173/? I/DEBUG:     #01  pc 00001265  /system/bin/linker (__dl_dlsym+325)
09-26 20:07:14.612 1173-1173/? I/DEBUG:     #02  pc 000bf16b  /data/app-lib/com.google.android.filament.textured-2/libfilament-jni.so (utils::ashmem_create_region(char const*, unsigned int)+43)
09-26 20:07:14.612 1173-1173/? I/DEBUG:     #03  pc fffffffe  <unknown>
```

Apparently we crash in the `dlsym` lookup we make in case we are on API 26. I could just skip the test on API 19 if I can query the API level at runtime from native.

File: filament/backend/src/opengl/OpenGLDriver.cpp [hunk 1]
```diff
@@ -573,20 +576,26 @@ void OpenGLDriver::textureStorage(OpenGLDriver::GLTexture* t,
...
-            // NOTE: if there is a mix of texture and renderbuffers, "fixed_sample_locations" must be true
-            // NOTE: what's the benefit of setting "fixed_sample_locations" to false?
+            if (TEXTURE_2D_MULTISAMPLE_SUPPORTED) {
+                // NOTE: if there is a mix of texture and renderbuffers, "fixed_sample_locations" must be true
+                // NOTE: what's the benefit of setting "fixed_sample_locations" to false?
 #if GLES31_HEADERS
-            // only supported from GL 4.3 and GLES 3.1
-            glTexStorage2DMultisample(t->gl.target, t->samples, t->gl.internalFormat,
-                    GLsizei(width), GLsizei(height), GL_TRUE);
+                // only supported from GL 4.3 and GLES 3.1
```
Comment (pixelflinger) [inline filament/backend/src/opengl/OpenGLDriver.cpp]: the comment about using a function-pointer is no longer valid, right?

Comment (romainguy) [reply filament/backend/src/opengl/OpenGLDriver.cpp]: Indeed, I removed it. I still need to fix all the dlsym crashes :(

**Comment by romainguy** [comment]
With the latest changes + a hack (commenting out all `dlsym()` calls in external texture support), I can run our sample apps on API level 19 in an emulator. The emulation layer is buggy and logs errors in the console but things work overall.

**Comment by romainguy** [comment]
Everything should now work on API 19.

File: filament/backend/src/opengl/PlatformEGL.cpp [hunk 2]
```diff
@@ -292,6 +299,9 @@ Driver* PlatformEGL::createDriver(void* sharedContext) noexcept {
 
     initializeGlExtensions();
 
+    // this is needed with older emulators/API levels on Android
```
Comment (pixelflinger) [inline filament/backend/src/opengl/PlatformEGL.cpp]: uh?! so some GL calls fail before we get here?

Comment (romainguy) [reply filament/backend/src/opengl/PlatformEGL.cpp]: On the emulator with ES 3.0 enabled and the KitKat system image, enumerating the GL extensions generates a GL error in the log (in the emulator driver, so not a bug on our side). ES 3.0 was just incomplete.

**Pull request merged**